### ISSUE TITLE
Support GDEF Ligature Caret List Table

### ISSUE THREAD

**Issue by Lohann** [RazrFalcon/ttf-parser#125]
I noticed that the **Ligature Caret List Table** parsing [is being skipped](https://github.com/RazrFalcon/ttf-parser/blob/v0.19.0/src/tables/gdef.rs#LL45C52-L45C52), I need this table in order to properly position the cursor and highlight text in a text editor when dealing with ligatures such as `ﬁ`, `ﬃ`, etc.
![image](https://github.com/RazrFalcon/ttf-parser/assets/4323004/7c7f8a85-67f6-4ef5-a501-1189e1ef8c15)

- https://learn.microsoft.com/en-us/typography/opentype/spec/gdef#ligature-caret-list-table-overview

Also seems that apple have its own custom ligature Caret Table called `lcar`:
- https://developer.apple.com/fonts/TrueType-Reference-Manual/RM06/Chap6lcar.html

**Comment by RazrFalcon** [comment]
Yes, this feature isn't supported by harfbuzz/rustybuzz, therefore there were no need in implementing it.

**Comment by daprahamian** [comment]
Hiya! It looks like I need the GDEF ligature caret list table as well. Would you be receptive to a PR implementing this?

**Comment by RazrFalcon** [comment]
Sure, patches are welcome.
